### ISSUE TITLE
'Duplicated Modules'

### ISSUE THREAD

**Issue by Harrisonl** [appsignal/appsignal-elixir-phoenix#4]
Running mix release on my elixir 1.10.4 application gives the following error:

```
** (Mix) Duplicated modules:
	'Elixir.Appsignal.Plug' specified in appsignal and appsignal_plug
	'Elixir.Appsignal.Phoenix' specified in appsignal and appsignal_phoenix
```

I'm sure I'm doing something wrong can't seem to figure it out. I think it might have something to do with the `Fallbacks` modules in the `appsignal` package.

If it makes a difference it IS an umbrella application. I have tried including only `appsignal_phoenix` and including that and root module. The only place I am calling `Appsignal` is in the endpoint file: `use Appsignal.Phoenix`

Might switch back to 1.13 for now and see how I get on but would be good to trial the beta!

**Comment by tombruijn** [comment]
Related support conversation (private link): https://app.intercom.com/a/apps/yzor8gyw/inbox/inbox/conversation/16410700008649#part_id=comment-16410700008649-6386555729

**Comment by jeffkreeftmeijer** [comment]
Hey @Harrisonl,

This sounds like an issue with the fallback modules we include in the main AppSignal package. We include a version of `Appsignal.Plug` and `Appsignal.Phoenix` if the `:appsignal_plug` or `:appsignal_phoenix` dependencies aren't present, so we can throw a warning to tell users to add them as dependencies.

However, these shouldn't be included in an actual release. We'll look into finding a way to keep them out of releases to fix this issue, and we'll be back with you once we know more. Thanks for letting us know about this!

**Comment by philipgiuliani** [comment]
Is there any workaround available?

**Comment by jeffkreeftmeijer** [comment]
Hey @Harrisonl and @philipgiuliani,

After looking into this some more, it looks like there's currently no proper way of doing module fallbacks. Thanks for reporting this! We're removing the fallbacks, and will be replacing them with a post-install message we'll print after installing the library to direct users who upgrade from 1.x to 2.x to the upgrade guide.

I've removed the fallback modules in the [`duplicate-modules` branch in appsignal-elixir](https://github.com/appsignal/appsignal-elixir/tree/duplicate-modules) , and we'll release a new version once we've added the post install message. For now, you can switch to that to fix this issue:

```elixir
defp deps do
  {:appsignal_phoenix, "~> 2.0.0-beta.1"},
  {:appsignal, github: "appsignal/appsignal-elixir", branch: "duplicate-modules", override: true},
end
```

**Comment by philipgiuliani** [comment]
Thank you. This resolved it for me 👍  Seems to work fine so far!

**Comment by jeffkreeftmeijer** [comment]
Great! Thanks for getting back to me. I'll work on the upgrade guide for the post-install message to link to before releasing a new version. I'll let you know when you can switch to a released version again here.

**Comment by jeffkreeftmeijer** [comment]
Hey all, we've just released version 2.0.0-beta.11 of the main package, which includes the fix for this issue. Could you upgrade to that from the `duplicate-modules` branch, and let us know if you run into issues like this again?

You should be able to remove any explicit dependencies on `:appsignal` if you're depending on `:appsignal_phoenix`, and run `mix deps.update appsignal` to get the new beta.
